packages/opencensus-core/src/trace/instrumentation/base-plugin.ts

```diff
@@ -54,16 +54,16 @@ export abstract class BasePlugin implements types.Plugin {
   }
 
 
-  // TODO: review this implementation
-  // From the perspective of an instrumentation module author,
-  // that applyUnpatch is abstract makes it seem like patching is optional,
-  // while unpatching is not. It should be the other way around
-
-  // tslint:disable:no-any
-  applyPatch(moduleExports: any, tracer: modelTypes.Tracer, version: string):
-      any {
-    this.setPluginContext(moduleExports, tracer, version);
-  }
+  /**
+   * Patches module exports functions.
+   * @param moduleExports nodejs module exports to be patched
+   * @param tracer a tracer instance to create spans from
+   * @param version module version description
+   */
+  abstract applyPatch(
+      // tslint:disable:no-any
+      moduleExports: any, tracer: modelTypes.Tracer, version: string): any;
```

Comment:
While we are here... maybe it makes sense to parameterize the return value. We know the `moduleExports` and the return value should be the same type, so we can make it `applyPatch<T>(moduleExports: T, ...): T;`.

Reply:
Makes a lot of sense. As `oc-core` changed since this PR was opened, I made this change to the `enable` method.